### PR TITLE
 Update VPP Device images for ARM 

### PR DESCRIPTION
**Pull request by jlinkes**
Added support for aarch64 VPP Device and made the current csit-shim and csit-sut Dockerfiles platform (x86_64 vs aarch64) agnostic.

### CODE REVIEW

File: csit-shim/Dockerfile [hunk 1]
```diff
@@ -1,30 +1,35 @@
...
-        && rm -rf /var/lib/apt/lists/*
+        ssh \
+        sudo \
+        tzdata \
+        uuid-runtime
 
-# Configure locales
-RUN locale-gen en_US 
+# Docker install
+RUN curl -fsSL https://get.docker.com | sh
```
Comment (pmikus) [inline csit-shim/Dockerfile]: Please make it a one command with above install. It will save one layer.

File: csit-shim/Dockerfile [hunk 2]
```diff
@@ -1,30 +1,35 @@
...
```
Comment (pmikus) [inline csit-shim/Dockerfile]: Please make it one command with line 15. It will save another layer.

File: csit-sut/Dockerfile [hunk 3]
```diff
@@ -23,27 +23,24 @@ ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en' LC_ALL='en_US.UTF-8'
...
-# Base layer
+# Configure locales
 RUN apt-get -q update \
  && apt-get install -y -qq \
+        locales
+RUN locale-gen en_US.UTF-8 \
+ && dpkg-reconfigure locales
+
+# Base layer
+RUN apt-get install -y -qq \
```
Comment (pmikus) [inline csit-sut/Dockerfile]: Merge apt-get layers into single RUN

File: csit-sut/Dockerfile [hunk 4]
```diff
@@ -116,13 +114,22 @@ RUN pip install \
...
 # VPP PIP pre-cache
 RUN pip install \
         aenum
 
+# Kernel image for VM tests download and extract
+COPY get-linux-vm-image.sh /root/get-linux-vm-image.sh
+RUN chmod 755 /root/get-linux-vm-image.sh
+RUN /root/get-linux-vm-image.sh
+
+RUN rm -rf /var/lib/apt/lists/*
```
Comment (pmikus) [inline csit-sut/Dockerfile]: Must be part of /root/get-linux-vm-image.sh script + should be also on line 40 of single apt-get

Comment (pmikus) [reply csit-sut/Dockerfile]: The problem is that this command has no effect. Every "RUN" is creating new Docker layer, if you do not make it clean in that layer, you lost access to clean it and resulting file will be huge as each apt-get is downloading lots of data.

**Comment by pmikus** [comment]
Just for making records. Do you have in plan or did you test those images? Sandbox?

**Comment by jlinkes** [comment]
> 
> 
> Just for making records. Do you have in plan or did you test those images? Sandbox?

I tested the arm images. I can test the x86_64 images if I have an environment for it.

File: csit-sut/Dockerfile [hunk 5]
```diff
@@ -116,12 +109,14 @@ RUN pip install \
...
-        scipy==1.1.0
+        numpy==1.14.5
 
-# VPP PIP pre-cache
-RUN pip install \
-        aenum
+# ARM workaround
+RUN pip install scipy==1.1.0
+
+# Kernel image for VM tests download and extract
```
Comment (pmikus) [inline csit-sut/Dockerfile]: Remove

File: csit-sut/Dockerfile [hunk 6]
```diff
@@ -11,10 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM ubuntu:bionic-20180821
-
+FROM ubuntu:18.04
```
Comment (pmikus) [inline csit-sut/Dockerfile:14]: Is this floating build or fixed? IoW will it change in time? Is so i prefer fixed like before.

File: csit-sut/get-linux-vm-image.sh [hunk 7]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env sh
```
Comment (pmikus) [inline csit-sut/get-linux-vm-image.sh]: remove?

File: csit-sut/Dockerfile (hunk 5 above)
Comment (jlinkes) [reply csit-sut/Dockerfile]: Right, we don't need the kernel image in csit-sut when it's on the host and accessible via mount point. I'll remove it.

File: csit-sut/get-linux-vm-image.sh (hunk 7 above)
Comment (jlinkes) [reply csit-sut/get-linux-vm-image.sh]: I'll remove the whole script since it's not needed.

**Comment by jlinkes** [comment]
Csit-sut has been tested in this patch:
https://gerrit.fd.io/r/#/c/20688/

And the actual run:
https://jenkins.fd.io/job/csit-vpp-device-master-ubuntu1804-1n-skx/2553/

**Comment by pmikus** [comment]
+1

**Review by pmikus — APPROVED** [review]
- Master image tested and works
- Jenkins executor not touched (ubuntu18)
- Old images cleaned

File: csit-sut/Dockerfile [hunk 8]
```diff
@@ -1,4 +1,5 @@
 # Copyright (c) 2019 Cisco and/or its affiliates.
+# Copyright (c) 2019 PANTHEON.tech s.r.o. and/or its affiliates.
```
Comment (snergfdio) [inline csit-sut/Dockerfile]: can't eat it with this....flip it back to cisco and i'll merge it straight away

Comment (jlinkes) [reply csit-sut/Dockerfile]: Removed.

After you merge it, could you build and push the images? And also setup csit-shim on the vpp device server?

**Comment by jlinkes** [comment]
Ed, is this ready for merging? If so, please build and push the images and install csit-shim on the tx2 (.69) node.

**Pull request merged**